docs/source/getting-started/app-architecture.rst

```diff
@@ -82,43 +82,66 @@ The UI
 On top of TypeScript, we use the UI framework `React <https://reactjs.org/>`_.
 React helps us write modular UI components using a functional style - a component is rerendered whenever one of its inputs changes - with careful use of global state.
 
-The latter is especially interesting as it's how we handle ledger state in our application.
-We use a state management feature of React called `Hooks <https://reactjs.org/docs/hooks-intro.html>`_.
-We use custom DAML React hooks to query the ledger for contracts, create new contracts, and exercise choices.
+Let's see an example of a React component.
+All components are in the ``ui/src/components`` folder.
+You can navigate there within Visual Studio Code using the file *Explorer* on the left sidebar.
+We'll first look at ``App.tsx``, which is the entry point to our application.
+
+.. literalinclude:: code/ui-before/App.tsx
+  :start-after: // APP_BEGIN
+  :end-before: // APP_END
+
+An important tool in the design of our components is a React feature called `Hooks <https://reactjs.org/docs/hooks-intro.html>`_.
+Hooks allow you to share and update state across components, avoiding having to thread it through manually.
```

Comment:
This might just be showing my ignorance of react but I thought that the thing that avoids having to thread it through are contexts not hooks.